lib/plugins/digging.js

```diff
@@ -162,6 +162,7 @@ function inject (bot) {
       bot.emit('diggingCompleted', newBlock)
       diggingTask.finish()
     }
+    bot.on('death', () => { bot.removeAllListeners('diggingAborted'); bot.removeAllListeners('diggingCompleted') })
```

Comment:
You should register the death event outside the dig function.

Reply:
Sorry, didn't realise.